model_bakery/random_gen.py

```diff
@@ -319,7 +320,10 @@ def gen_geometry_collection() -> str:
 
 def gen_pg_numbers_range(number_cast: Callable[[int], Any]) -> Callable:
     def gen_range():
-        from psycopg2._range import NumericRange
+        if django.VERSION >= (4, 2):
+            from psycopg.types.range import NumericRange
+        else:
+            from psycopg2._range import NumericRange
```

Comment:
I wonder if that is the best option here.

I see that Django now supports psycopg3, but does not default to it (I mean, you are free to pick whatever database driver and connection you want, right?). It still supports psycopg2 as well and not deprecated yet:
https://docs.djangoproject.com/en/4.2/releases/4.2/#psycopg-3-support

So, I'd rather try to detect psycopg version installed and decide on the imports based on this.
If there is no clean way of detecting the installed version (which I doubt), then we can fall back to the catching `ImportError`, defaulting to the new psycopg.

What do you think?

Reply:
Hmm you're right. The reason I went with this approach is that it makes it easier to find these later (a search for `if django.VERSION` whenever a Django version is deprecated makes removing this kind of dead code much easier). However since there isn't (yet) a deprecation path for pyscopg2 in Django I agree this makes sense. I'll change it.

Reply:
Thanks!